### PR TITLE
fix(scaleway): use jsonify for application/json content-type only

### LINKED ISSUE #65036: scaleway_user_data: multiline is broken, cloud-init doesn't work
##### SUMMARY
There should be multiline in the cloud-init field, but instead either the module or API itself breaks it and sends the whole text as a string in double quotes.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
scaleway_user_data

##### ANSIBLE VERSION

```paste below
ansible 2.8.3
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

```yaml
  - scaleway_user_data:
      server_id: "{{ scaleway_compute.msg.id }}"
      region: "{{ region }}"
      user_data:
        cloud-init: |
          #cloud-config
          # final_message
          # default: cloud-init boot finished at $TIMESTAMP. Up $UPTIME seconds
          # this message is written by cloud-final when the system is finished
          # its first boot
          final_message: "The system is finally up, after $UPTIME seconds"
```

1. Update the cloud-init and start the server
2. Cloud-init throws an error
`2019-11-18 19:27:31,692 - __init__.py[WARNING]: Unhandled non-multipart (text/x-not-multipart) userdata: 'b'"#cloud-config\\\\nfinal_me'...'`

##### EXPECTED RESULTS

```
root@test-cloudinit:~# curl --local-port 1-1023 http://169.254.42.42/user_data/cloud-init/ -w "\n"
#cloud-config
# final_message
# default: cloud-init boot finished at $TIMESTAMP. Up $UPTIME seconds
# this message is written by cloud-final when the system is finished
# its first boot
final_message: "The system is finally up, after $UPTIME seconds"
```

##### ACTUAL RESULTS

```
root@test-cloudinit:~# curl --local-port 1-1023 http://169.254.42.42/user_data/cloud-init/ -w "\n"
"#cloud-config\n# final_message\n# default: cloud-init boot finished at $TIMESTAMP. Up $UPTIME seconds\n# this message is written by cloud-final when the system is finished\n# its first boot\nfinal_message: \"The system is finally up, after $UPTIME seconds\"\n"
```

### PR DESCRIPTION
**Pull request by acamilleri**
##### SUMMARY
by default, the data is not modified and is sent as specified unless the content type is application / json.

Fixes #65036

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
`lib/ansible/module_utils/scaleway.py`

##### ADDITIONAL INFORMATION
I'm using the [scaleway_user_data](https://docs.ansible.com/ansible/latest/modules/scaleway_user_data_module.html) module, but i can't passed a multi-line configuration, because it's convert to json format.

##### EXPECTED RESULT
```
$ curl -XGET -H "X-Auth-Token: $SCW_SECRET_TOKEN" https://cp-par1.scaleway.com/servers/$SCW_SERVER_ID/user_data/cloud-init

#cloud-config
apt_update: true
write_files:
- path: /etc/ssh/sshd_config
  content: |
    Port 55555
    Protocol 2
    PasswordAuthentication no
    ChallengeResponseAuthentication no
    UsePAM no
    X11Forwarding yes
    PrintMotd no
    AcceptEnv LANG LC_*
    Subsystem	sftp	/usr/lib/openssh/sftp-server
```

##### STEPS TO REPRODUCE
Task:
```
  - name: set cloud-init user-data
    scaleway_user_data:
      api_url: "https://api.scaleway.com/instance/v1/zones/fr-par-1"
      api_token: "{{ lookup('env','SCW_SECRET_KEY') }}"
      server_id: "{{ lookup('env','SCW_SERVER_ID') }}"
      region: "par1"
      user_data:
        cloud-init: |
          #cloud-config
          apt_update: true
          write_files:
          - path: /etc/ssh/sshd_config
            content: |
              Port 2222
              Protocol 2
              PasswordAuthentication no
              ChallengeResponseAuthentication no
              UsePAM no
              X11Forwarding yes
              PrintMotd no
              AcceptEnv LANG LC_*
              Subsystem	sftp	/usr/lib/openssh/sftp-server
          runcmd:
          - systemctl restart sshd
```

Unexpected result:
```
$ curl -XGET -H "X-Auth-Token: $SCW_SECRET_TOKEN" https://cp-par1.scaleway.com/servers/$SCW_SERVER_ID/user_data/cloud-init

"#cloud-config\napt_update: true\nwrite_files:\n- path: /etc/ssh/sshd_config\n  content: |\n    Port 55555\n    Protocol 2\n    PasswordAuthentication no\n    ChallengeResponseAuthentication no\n    UsePAM no ..."
```

Cloud-init not worked:
```
2020-01-30 11:02:49,625 - __init__.py[WARNING]: Unhandled non-multipart (text/x-not-multipart) userdata: 'b'"#cloud-config\\\\napt_upda'...'
```

### CODE REVIEW

File: lib/ansible/module_utils/scaleway.py [hunk 1]
```diff
@@ -118,11 +118,14 @@ def _url_builder(self, path, params):
     def send(self, method, path, data=None, headers=None, params=None):
         url = self._url_builder(path=path, params=params)
         self.warn(url)
-        data = self.module.jsonify(data)
+        data = data
```
Comment (remyleone) [inline lib/ansible/module_utils/scaleway.py]: Why are you doing a `data = data`

Comment (acamilleri) [reply lib/ansible/module_utils/scaleway.py]: It's a mistake, i removed it.

**Comment by remyleone** [comment]
shipit

**Comment by remyleone** [comment]
LGTM

**Comment by gundalow** [comment]
rebuild_merge

**Comment by felixfontein** [comment]
Looks like there is something wrong with shippable, the test is still "PROCESSING" even though it is [done](https://app.shippable.com/github/ansible/ansible/runs/159565/summary/console).

**Comment by felixfontein** [comment]
BTW, this is missing a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

**Comment by acamilleri** [comment]
@felixfontein i added changelog, ok for you ?

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@acamilleri thanks for fixing this!
@remyleone thanks for reviewing!

**Pull request merged**